src/cli/commands/ls.js

```diff
@@ -48,11 +48,7 @@ module.exports = {
       const multihashWidth = Math.max.apply(null, links.map((file) => file.hash.length))
       const sizeWidth = Math.max.apply(null, links.map((file) => String(file.size).length))
 
-      let pathParts = key.split('/')
-
-      if (key.startsWith('/ipfs/')) {
-        pathParts = pathParts.slice(2)
-      }
+      let pathParts = key.replace(/\/ipfs\/|\/$/g, '').split('/')
```

Comment:
What would happen when we try to list paths that contain `ipfs` inside UnixFS DAGs?  E.g. `jsipfs ls /ipfs/Qmfoo/ipfs/bar`?

Reply:
Good catch!

Reply:
Regular expressions, now you have two problems, etc, etc 🤣